projects/packages/waf/src/class-brute-force-protection.php

```diff
@@ -547,11 +547,16 @@ public function log_failed_attempt( $username, $error = null ) {
 		if ( isset( $_COOKIE['jpp_math_pass'] ) ) {
 
 			$transient = $this->get_transient( 'jpp_math_pass_' . sanitize_key( $_COOKIE['jpp_math_pass'] ) );
-			--$transient;
+			if ( $transient ) {
+				--$transient;
+			}
```

Comment:
Not a blocker, and probably not worth spending too much time on, but...

It seems `false` is definitely a possibility, as `$this->get_transient()` is a wrapper for `get_transient()`, which returns `false` if the value is not found. In that scenario, there's probably no need to delete the transient.

In weird setups, this fix will still throw a deprecation if `$transient` is `true` or `'bfp'`, or a fatal with `array(1)`. We could do an `is_int()` check to avoid that possibility.

Reply:
A very good thought indeed, better safe than sorry!

I added the check and after adding a test to cover it noticed that the `if` below this one also failed in some non-integer cases, so I extended it to play nice with invalid values as well. All of that is in a018a9d7e4f15d49bb9c29b5624e867b5a1febc5.